packages/eslint-plugin/src/rules/no-implied-eval.ts

```diff
@@ -80,6 +80,16 @@ export default util.createRule({
         return true;
       }
 
+      if (symbol && symbol.escapedName === FUNCTION_CONSTRUCTOR) {
```

Comment:
If don't check 
```

// symbol.escapedName === 'Function'
symbol.escapedName === FUNCTION_CONSTRUCTOR
```
then this TC failed. because return true.
```
const foo = new String('x + 1');

setTimeout(foo, 0);
setInterval(foo, 0);
setImmediate(foo);
execScript(foo);
```
Please let me know if you have a better idea 😃